### ISSUE TITLE
[BUG] npm install arborist error on create react app 

### ISSUE THREAD

**Issue by ruyadorno** [npm/cli#2092]
### Current Behavior:
Trying to run `npm install` in a `create-react-app` generated project and adding peer dependencies to top-level ends with unexpected failure:

**package.json**
```json
{
  "name": "newapp",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "^5.11.4",
    "@testing-library/react": "^11.1.0",
    "@testing-library/user-event": "^12.1.10",
    "react-dom": "^17.0.1",
    "react-scripts": "4.0.0",
    "web-vitals": "^0.2.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "peerDependencies": {
    "ink": "^3.0.8",
    "react": "^17.0.1"
  }
}
```

```
npm verb stack TypeError: Cannot read property 'matches' of undefined
npm verb stack     at Arborist.[canPlaceDep] (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1288:15)
npm verb stack     at Arborist.[placeDep] (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1065:36)
npm verb stack     at Arborist.[placeDep] (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:1216:41)
npm verb stack     at /Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:767:46
npm verb stack     at Array.map (<anonymous>)
npm verb stack     at Arborist.[buildDepStep] (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:767:8)
npm verb stack     at processTicksAndRejections (node:internal/process/task_queues:93:5)
npm verb stack     at async Arborist.buildIdealTree (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/build-ideal-tree.js:203:7)
npm verb stack     at async Promise.all (index 1)
npm verb stack     at async Arborist.reify (/Users/ruyadorno/Documents/workspace/cli/main/node_modules/@npmcli/arborist/lib/arborist/reify.js:121:5)
```

### Expected Behavior:
`npm install` should complete install.

```
"peerDependencies": {
    "ink": "^3.0.8",
    "react": "^17.0.1"
  }
```

### Steps To Reproduce:
Create new project with `create-react-app`: 

- `npx create-react-app newfolder`
- `cd newfolder`
- `npm install`

### Environment:

- OS: MacOS 10.15.7
- Node: 15.0.1
- npm: 7.0.7

**Comment by isaacs** [comment]
Ah, yeah, this is failing because there's a conflict on the `peerOptional` dep.  It should fail, but should fail with this error:

```
Error: unable to resolve dependency tree
    at Arborist.[failPeerConflict] (/Users/isaacs/dev/npm/arborist/lib/arborist/build-ideal-tree.js:1020:25)
    at Arborist.[placeDep] (/Users/isaacs/dev/npm/arborist/lib/arborist/build-ideal-tree.js:1091:30)
    at /Users/isaacs/dev/npm/arborist/lib/arborist/build-ideal-tree.js:772:46
    at Array.map (<anonymous>)
    at Arborist.[buildDepStep] (/Users/isaacs/dev/npm/arborist/lib/arborist/build-ideal-tree.js:772:8)
    at async Arborist.buildIdealTree (/Users/isaacs/dev/npm/arborist/lib/arborist/build-ideal-tree.js:203:7)
    at async Promise.all (index 1)
    at async Arborist.reify (/Users/isaacs/dev/npm/arborist/lib/arborist/reify.js:121:5) {
  code: 'ERESOLVE',
  current: {
    name: 'type-fest',
    version: '0.12.0',
    location: 'node_modules/type-fest',
    dependents: [
      {
        type: 'prod',
        name: 'type-fest',
        spec: '^0.12.0',
        from: {
          name: 'ink',
          version: '3.0.8',
          location: 'node_modules/ink',
          dependents: [
            {
              type: 'peer',
              name: 'ink',
              spec: '^3.0.8',
              from: { location: '/Users/isaacs/dev/npm/cli/myapp' }
            }
          ]
        }
      }
    ]
  },
  edge: {
    type: 'peerOptional',
    name: 'type-fest',
    spec: '^0.13.1',
    error: 'INVALID',
    from: {
      name: '@pmmmwh/react-refresh-webpack-plugin',
      version: '0.4.2',
      location: 'node_modules/@pmmmwh/react-refresh-webpack-plugin',
      dependents: [
        {
          type: 'prod',
          name: '@pmmmwh/react-refresh-webpack-plugin',
          spec: '0.4.2',
          from: {
            name: 'react-scripts',
            version: '4.0.0',
            location: 'node_modules/react-scripts',
            dependents: [
              {
                type: 'prod',
                name: 'react-scripts',
                spec: '4.0.0',
                from: { location: '/Users/isaacs/dev/npm/cli/myapp' }
              }
            ]
          }
        }
      ]
    }
  },
  peerConflict: null,
  strictPeerDeps: false,
  force: false
}
```

**Comment by isaacs** [comment]
What's happening is that we don't have a valid resolution, so we try to pull from the virtualRoot.  But, the virtualRoot doesn't have it, because we don't load optional peers by default.  So the correct thing to do is leave it as an unmet dependency at that point, and attempt to resolve it later in the process.

**Comment by ruyadorno** [comment]
Fixed in `v7.0.8`
